### PR TITLE
Add key manager for managing private keys

### PR DESCRIPTION
**Pull request by ishbir**
Package keymgr is a key manager for use with bmclient. However, it is meant to be used independently and handles hierarchically deterministic keys, imported keys (from PyBitmessage or other legacy systems), as well as channels.

The file format for storage is a JSON document encrypted with SalsaX20 and MAC'd with Poly1305. The scheme used is byte compatible with secretbox from NaCl. Currently, it supports encrypting the entire file containing identity information (encryption and signing keys, required PoW constants, etc) and master key.

Future support for separately encrypting master key and signing keys is planned. This could be useful for:
* Read only bmclient nodes that are intended to only receive and decrypt messages bound for them and are not likely to be used for sending messages.
* Security conscious users who might be worried about memory-reading malware. Note that even if we configure bmclient as such, the malware would still be able to get the user's private encryption key. By the very nature of Bitmessage, private encryption keys need to be in memory all the time.

The contents of the encrypted file are arranged as such: `Nonce (24 bytes) || Salt for PBKDF2 (32 bytes) || Encrypted data`

Both the nonce and salt are re-generated each time the file needs to be saved. This, along with a high number of PBKDF2 iterations (2^15), helps to ensure that an adversary with access to the key file will have an extremely hard time trying to bruteforce it. Rainbow tables will be useless.

WARNING (again): The key manager is insecure against memory reading malware and is at the mercy of Go's garbage collector.

PR depends on: https://github.com/monetas/bmutil/pull/18

### CODE REVIEW

File: log.go [hunk 1]
```diff
@@ -68,14 +62,6 @@ func useLogger(subsystemID string, logger btclog.Logger) {
...
 	case "RPC":
 		rpcLog = logger
-
-	case "KEYMGR":
-		keymgrLog = logger
-		keymgr.UseLogger(logger)
-
-	case "MSGDB":
-		msgdbLog = logger
-		msgdb.UseLogger(logger)
```
Comment (DanielKrawisz) [inline log.go]: Why did you remove the key manager log?

Comment (ishbir) [inline log.go]: Because it wasn't needed anymore.
